CriticalMass/URLCodable.swift

```diff
@@ -46,28 +46,34 @@ extension URLCodable {
         return try self.init(scheme: urlComponents.scheme, host: urlComponents.host, path: urlComponents.path, queryObject: codableObject)
     }
 
-    private func encode<T: Encodable>(codable: T) throws -> [URLQueryItem] {
+    private static func decode<T: Codable>(type: T.Type, from items: [URLQueryItem]) throws -> T {
+        let dict = items.reduce(into: [String: String]()) { result, item in
+            result[item.name] = item.value
+        }
+        let data = try JSONSerialization.data(withJSONObject: dict, options: [])
+        return try JSONDecoder().decode(type, from: data)
+    }
+}
+
+extension URLQueryItem {
+    static func encode<T: Encodable>(codable: T) throws -> [URLQueryItem] {
         let jsonData = try JSONEncoder().encode(codable)
```

Comment:
could be:
```suggestion
        let jsonData = try codable.encoded()
```

Reply:
Nice that helper method also enables us to remove the QueryDefiningDataType workaround 🎉